Frontend/src/App.tsx

```diff
@@ -113,12 +132,61 @@ function App() {
   const downloadPackage = async () => {
     if (selectedPackage) {
       try {
-        await axios.get(`/package/${selectedPackage}`);
+        const response = await axios.get(`/package/${selectedPackage}`);
+        const base64Content = response.data;
+        // Convert base64 to binary
+        const binaryContent = atob(base64Content);
+        // Create a Uint8Array from the binary data
+        const uint8Array = new Uint8Array(binaryContent.length);
+        for (let i = 0; i < binaryContent.length; i++) {
+          uint8Array[i] = binaryContent.charCodeAt(i);
+        }
+        // Create a blob from the Uint8Array
+        const blob = new Blob([uint8Array], { type: 'application/zip' });
+        // Create a link element and simulate a click to trigger the download
+        const link = document.createElement('a');
+        link.href = window.URL.createObjectURL(blob);
+        link.download = `${selectedPackage}.zip`; // Set the desired file name
+        link.click();
       } catch (error: any) {
         console.error(`Error downloading package: ${error.message}`);
       }
     }
   };
+  
+  const handleUpdateClick = () => {
+    const content = prompt('Enter new content:');
+    const url = prompt('Or enter update url:');
+
+    if ((content || url) && !(content && url)) {
+      setUpdateFields({ content, url });
+      if(url){
+        axios.put(`/package/${selectedPackage}`, { url });
+      }
+      else{
+        axios.put(`/package/${selectedPackage}`, { content });
+      }
+    }
+    else{
+      alert('Please enter either content or url');
+    }
+  };
+
+  // Rating functionality
+  const handleRatingClick = async () => {
+    const packageId = prompt('Enter package ID:');
+
+    if (packageId) {
+      const response = await axios.get(`/package/${packageId}/rate`);//Make API call to get ratings based on packageId
+
+      if ((response).status === 200) {
+        const ratings = response.data;
+        setPackageRating(ratings);
+      } else {
+        alert('Unable to retrieve ratings for the specified package ID.');
```

Comment:
you should account for each of the error codes you get instead of a catch all for non 200 responses. Just do a switch statement on the response.status and do an alert based on the error code you get.